frontend/packages/core/src/NPS/header.tsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import {
+  ClickAwayListener,
+  Grow as MuiGrow,
+  MenuList,
+  Paper as MuiPaper,
+  Popper as MuiPopper,
+} from "@material-ui/core";
+import ChatBubbleOutlineIcon from "@material-ui/icons/ChatBubbleOutline";
+import { sortBy } from "lodash";
+
+import type { Workflow } from "../AppProvider/workflow";
+import { IconButton } from "../button";
+import { useAppContext } from "../Contexts";
+import type { SelectOption } from "../Input";
+import styled from "../styled";
+
+import NPSFeedback from "./feedback";
+
+const Grow = styled(MuiGrow)((props: { placement: string }) => ({
+  transformOrigin: props.placement,
+}));
+
+const Popper = styled(MuiPopper)({
+  padding: "0 12px",
+  marginLeft: "12px",
+  zIndex: 1201,
+});
+
+const Paper = styled(MuiPaper)({
+  width: "350px",
+  border: "1px solid #E7E7EA",
+  boxShadow: "0px 15px 35px rgba(53, 72, 212, 0.2)",
+  borderRadius: "8px",
+});
+
+const StyledFeedbackIcon = styled(IconButton)<{ $open: boolean }>(
+  {
+    color: "#ffffff",
+    marginRight: "8px",
+    padding: "12px",
+    "&:hover": {
+      background: "#2d3db4",
+    },
+    "&:active": {
+      background: "#2938a5",
+    },
+  },
+  props => ({
+    background: props.$open ? "#2d3db4" : "unset",
+  })
+);
+
+export const generateFeedbackTypes = (workflows: Workflow[]): SelectOption[] => {
+  const feedbackTypes: SelectOption[] = [{ label: "General" }];
+
+  workflows.forEach(workflow => {
+    feedbackTypes.push({
+      label: workflow.group,
+      group: sortBy(
+        workflow.routes.map(route => ({
+          label: route.displayName,
+          value: `/${workflow.path}/${route.path}`,
+        })),
+        ["label"]
+      ),
+    });
+  });
+
+  return feedbackTypes;
+};
+
+const HeaderFeedback = () => {
+  const [open, setOpen] = React.useState<boolean>(false);
+  const anchorRef = React.useRef(null);
+  const { workflows } = useAppContext();
+
+  const handleToggle = () => {
+    setOpen(!open);
+  };
+
+  const handleClose = event => {
+    // handler so that it wont close when selecting an item in the select
+    if (event.target.localName === "body") {
+      return;
+    }
+    if (anchorRef.current && anchorRef.current.contains(event.target)) {
+      return;
+    }
+    setOpen(false);
+  };
+
+  const FeedbackIconProps = {
```

Comment:
just curious, why not inline these?

Reply:
Not sure why I left it that way, good call though, inlining and removing